### PR TITLE
[IDP-1496] Add rules to enforce short schema name to support ruby client generation

### PR DESCRIPTION
**Pull request by PrincessMadMath**
## Description of changes
New rules that check the lenght of schema.

Required for 

## Breaking changes
New rules, so maybe new infraction in existing code base.

## Additional checks

- [x] Added new tests that cover the code changes

### CODE REVIEW

File: .spectral.yaml [hunk 1]
```diff
@@ -102,3 +102,11 @@ rules:
       function: pattern
       functionOptions:
         notMatch: 'string'
+
+  schema-name-length:
```
Comment (heqianwang) [inline .spectral.yaml]: I think we can have a better name which indicates why its being flagged:
something like `schema-name-too-long` or `schema-name-must-be-shorter`?

File: .spectral.yaml [hunk 2]
```diff
@@ -102,3 +102,11 @@ rules:
...
+    description: "Schema name must not be too long to support client generation. Current limitation come from the Ruby using tar which have a limit of 100 characters."
+    given: $.components.schemas[*]~
+    then:
+      function: length
+      functionOptions:
+        max: 95
```
Comment (heqianwang) [inline .spectral.yaml:112]: why 95 over 99 or 100?

**Review by heqianwang — APPROVED** [review]

File: .spectral.yaml (hunk 1 above)
Comment (PrincessMadMath) [reply .spectral.yaml]: Good points

File: .spectral.yaml (hunk 2 above)
Comment (PrincessMadMath) [reply .spectral.yaml:112]: Real limit is 100, minus the extensino + small buffer just to be safe.

**Review by heqianwang — APPROVED** [review]

File: .spectral.yaml [hunk 3]
```diff
@@ -102,3 +102,11 @@ rules:
       function: pattern
       functionOptions:
         notMatch: 'string'
+
+  schema-name-length-must-be-short:
+    description: "Schema name must not be too long to support client generation. Current limitation come from the Ruby using tar which have a limit of 100 characters."
```
Comment (heqianwang) [inline .spectral.yaml]:
```suggestion
    description: "Schema name must not be too long to support client generation. Current limitation comes from Ruby packages which uses tar and has a limit of 100 characters."
```

**Review by heqianwang — APPROVED** [review]

**Review by heqianwang — APPROVED** [review]

**Pull request merged**